tests/js/common/http/grants-spec.js

```diff
@@ -134,3 +135,209 @@ describe('Grants', function() {
+  };
+
+  const generateConfigBody = (object) => {
+    return {
+      value: object
+    };
+  };
+
+  before(function () {
+    db._createDatabase(nonRootDB);
+  });
+
+  beforeEach(function () {
+    try {
+      // remove any such user if it exists
+      users.remove(nonRootUser);
+    } catch (ignore) {
+    }
+  });
+
+  afterEach(function () {
+    try {
+      users.remove(nonRootUser);
+    } catch (ignore) {
+    }
+  });
+
+  after(function () {
+    db._dropDatabase(nonRootDB);
+  });
+
+  const testSuiteHelper = (user, database) => {
+    const readUrl = generateReadConfigUrl(user, database);
+    const writeUrl = generateWriteConfigUrl(user, database);
+    let response;
+
+    // should be empty by default
+    response = request.get({
+      url: readUrl,
+      json: true
+    });
+    verifyEmptyConfig(JSON.parse(response.body));
+
+    // stores the init object
+    response = request.put({
+      url: writeUrl,
+      body: generateConfigBody(initConfigObject),
+      json: true
+    });
+    expect(response.statusCode).to.equal(200);
+
+    // should now be the same as init object
+    response = request.get({
+      url: readUrl,
+      json: true
+    });
+    verifyInitObject(JSON.parse(response.body));
+
+    // now try to overwrite as well
+    response = request.put({
+      url: writeUrl,
+      body: generateConfigBody(overwriteConfigObject),
+      json: true
+    });
+    expect(response.statusCode).to.equal(200);
+
+    response = request.get({
+      url: readUrl,
+      json: true
+    });
+    verifyOverwriteConfigObject(JSON.parse(response.body));
+
+    // now try a merge with another storage attribute
+    const additionalWriteUrl = generateWriteConfigUrl(user, database, additionalPropertyAttribute);
+    response = request.put({
+      url: additionalWriteUrl,
+      body: generateConfigBody(mergeConfigObject),
+      json: true
+    });
+    expect(response.statusCode).to.equal(200);
+    response = request.get({
+      url: readUrl,
+      json: true
+    });
+    verifyMergedConfigObject(JSON.parse(response.body));
+
+    // test delete (remove additional key)
+    response = request.delete({
+      url: additionalWriteUrl,
+      json: true
+    });
+    response = request.get({
+      url: readUrl,
+      json: true
+    });
+    verifyOverwriteConfigObject(JSON.parse(response.body));
+
+    // finally, test delete of graph attribute
+    response = request.delete({
+      url: writeUrl,
+      json: true
+    });
+    // should be fully empty now again
+    response = request.get({
+      url: readUrl,
+      json: true
+    });
+    verifyEmptyConfig(JSON.parse(response.body));
+  };
+
+  it('should store/patch/delete root config data in _system database', function () {
+    testSuiteHelper(rootUser, rootDB);
+  });
+
+  it('should store/patch/delete root config data in non-system database', function () {
+    testSuiteHelper(rootUser, nonRootDB);
```

Comment:
Side node: there is no cleanup for the root user's data.
The test relies on that the cleanup in line 310 works. If it ever fails, the second test that is testing the root user's data will see old values and may return false results.

Reply:
Thanks. Also added in: b481f7a013da7b072ec87d13740ed41d1dfb4469